Alertinator.php

```diff
@@ -68,7 +83,76 @@ public function check() {
          }
       }
    }
+   
+   /**
+    * Threshold notification: determine if an all-clear alert should be sent,
+    * and if so, send it and reset the logger.
+   */
+   private function notifyClear($check, $alertAfter, $clearAfter, $alerteeGroups) {
+      $log = $this->logger->readAlerts($check);
+      krsort($log);
+      
+      // We only get here if there was at least 1 failure, but 1 failure may not
+      // exceed the alertAfter threshold. If the check succeeds without
+      // reaching the alert threshold, reset the log silently.
+      // TODO: this will break terribly for bouncing errors, e.g. pass-> fail->
```

Comment:
Please add an issue for any TODO and mark that issue number in the TODO comment

Reply:
Good catch